purchase_analytic_global/models/purchase_order.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2014-2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+import json
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class PurchaseOrder(models.Model):
+    _name = "purchase.order"
+    _inherit = ["purchase.order", "analytic.mixin"]
+
+    analytic_distribution = fields.Json(
+        inverse="_inverse_analytic_distribution",
+        help="This analytic distribution will be propagated to all lines "
+        "analytic distributions, if you need to use different "
+        "analytic distribution, define it at line level.",
+    )
+
+    @api.depends("order_line.analytic_distribution")
+    def _compute_analytic_distribution(self):
+        """Set the analytic distribution on the order based on its order lines.
+
+        If all order lines have the same analytic distribution,
+        then set it on the order, otherwise left the field empty.
+        """
+        res = None
+        if hasattr(super(), "_compute_analytic_distribution"):
+            res = super()._compute_analytic_distribution()
```

Comment:
```suggestion
        res = super()._compute_analytic_distribution()
```

it's defined in the mixin, so you can be sure it's there :)